### ISSUE TITLE
Test failure: ClosingConnectionMidChunkPrefixThrows

### ISSUE THREAD

**Issue by aspnet-hello** [aspnet/KestrelHttpServer#2883]
This test [fails](http://aspnetci/viewLog.html?buildId=544428&buildTypeId=Releases_21Public_UbuntuUniverse) occasionally with the following error:
```
Assert.Equal() Failure
          ↓ (pos 0)
Expected: 
Actual:   HTTP/1.1 200 OK\r\nDate: Tue, 04 Sep 2018 2···
          ↑ (pos 0)
   at Microsoft.AspNetCore.Testing.TestConnection.ReceiveEnd(String[] lines) in /_/test/shared/TestConnection.cs:line 159
   at Microsoft.AspNetCore.Server.Kestrel.FunctionalTests.ChunkedRequestTests.ClosingConnectionMidChunkPrefixThrows(ListenOptions listenOptions) in /_/test/Kestrel.FunctionalTests/ChunkedRequestTests.cs:line 683
--- End of stack trace from previous location where exception was thrown ---
```
Other tests within that build may have failed with a similar message, but they are not listed here. Check the link above for more info.

This test failed on 2.1.

CC @muratg

This issue was made automatically. If there is a problem contact ryanbrandenburg.

**Comment by aspnet-hello** [comment]
Please use this workflow to address this flaky test issue, including checking applicable checkboxes and filling in the applicable "TODO" entries:

* Is this actually a flaky test?
  * No, this is a regular test failure, fix the test/product (TODO: Link to commit/PR)
  * Yes, proceed below...

* Is this test failure caused by product code flakiness? (Either this product, or another product this test depends on.)
  * [ ] File a bug against the product (TODO: Link to other bug)
  * Is it possible to change the test to avoid the flakiness?
    * Yes? Go to the "Change the test!" section.
    * No?
      * [ ] Disable the test (TODO: Link to PR/commit)
      * [ ] Wait for other bug to be resolved
      * [ ] Wait for us to get build that has the fix
      * [ ] Re-enable our test (TODO: Link to PR/commit)
      * [ ] Close this bug

* Is it that the test itself is flaky? This includes external transient problems (e.g. remote server problems, file system race condition, etc.)
  * Is there is a way to change our test to avoid this flakiness?
    * Yes? Change the test!
      * [ ] Change the test to avoid the flakiness, for example by using a different test strategy, or by adding retries w/ timeouts (TODO: Link to PR/commit)
      * [ ] Run the test 100 times locally as a sanity check.
      * [ ] Close this bug
    * No?
      * Is there any logging or extra information that we could add to make this more diagnosable when it happens again?
        * Yes?
            * [ ] Add the logging (TODO: Link to PR/commit)
        * No?
            * [ ] Delete the test because flaky tests are not useful (TODO: Link to PR/commit)

This comment was made automatically. If there is a problem contact ryanbrandenburg.

**Comment by muratg** [comment]
cc @JunTaoLuo @Tratcher @halter73

**Comment by JunTaoLuo** [comment]
This needs a back port of src changes to 2.1: https://github.com/aspnet/KestrelHttpServer/pull/2734. cc @muratg for 2.1 shiproom considerations.

**Comment by muratg** [comment]
@Eilon Another candidate for 2.1.5 (or perhaps 2.1.6).

**Comment by Eilon** [comment]
We should spend some time to figure out if there's a way to make the test more reliable without changing the product code (i.e. anything in src/). This would be a very tough sell to include in LTS unless customers have reported this issue and are blocked.

**Comment by Eilon** [comment]
@halter73 I'm fine with something like that so long as we don't feel that the reduced coverage would likely mask a future bug. We'll still have the full coverage in 2.2 and newer anyway. And 2.1.x is extremely stable in general because there are very few changes going into it.

**Comment by JunTaoLuo** [comment]
Test fix in 2.1 is in. No src change made.
